go-controller/pkg/util/pod_annotation.go

```diff
@@ -339,6 +309,49 @@ func GetPodIPsOfNetwork(pod *v1.Pod, nInfo NetInfo) ([]net.IP, error) {
 	return []net.IP{ip}, nil
 }
 
+func SecondaryNetworkPodIPs(pod *v1.Pod, networkInfo NetInfo) ([]net.IP, error) {
+	ips := []net.IP{}
+	podNadNames, err := PodNadNames(pod, networkInfo)
+	if err != nil {
+		return nil, err
+	}
+	for _, nadName := range podNadNames {
+		ips = append(ips, getAnnotatedPodIPs(pod, nadName)...)
+	}
+	return ips, nil
+}
+
+func PodNadNames(pod *v1.Pod, netinfo NetInfo) ([]string, error) {
+	on, networkMap, err := GetPodNADToNetworkMapping(pod, netinfo)
+	// skip pods that are not on this network
+	if err != nil {
+		return nil, err
+	} else if !on {
+		// if we are not able to determine if this pod is on this network continue processing the
+		// remaining pods anyway, hopefully the pod will be handled in a future pod update event.
+		klog.Warningf("Failed to determine if pod %s/%s is on network %s: %v",
+			pod.Namespace, pod.Name, netinfo.GetNetworkName(), err)
```

Comment:
I would consider not having this warning here. If at all possible, I would expect the user of the function to log if not satisfied with the results. From the perspective that this is an utility function a user can potentially use without being concerned about this.
Not strong opinion, I have the tendency to avoid error/warning logging from utility functions as much as possible.

Reply:
yes good point, i will fix